adb_graphics/datahandler/gribdata.py

```diff
@@ -258,217 +153,122 @@ def get_transform(self, transforms, val):
-        lev_val = ''.join([c for c in level if (c in digits or c == '.')])
-
-        # Convert the numbers to a list, and make integers or floats
-        if lev_val:
-            if split is not None:
-                lev_val = [int(lev) for lev in lev_val]
-            else:
-                lev_val = [float(lev_val) if '.' in lev_val else int(lev_val)]
-
-        # Gather all the letters
-        lev_unit = ''.join([c for c in level if c in ascii_letters])
-
-        if index_match:
-            if lev_unit == 'cm':
-                lev_val = [val / 100. for val in lev_val]
-            if lev_unit in ['mb', 'mxmb']:
-                lev_val = [val * 100. for val in lev_val]
-            if lev_unit in ['in', 'km', 'mn', 'mx', 'sr']:
-                lev_val = [val * 1000. for val in lev_val]
-
-        return lev_val, lev_unit
+    def latlons(self) -> list[np.ndarray]:
+        """Returns the set of latitudes and longitudes."""
+
+        coords = sorted(
+            [
+                str(c)
+                for c in list(self.field.coords)
+                if any(ele in str(c) for ele in ["lat", "lon"])
+            ]
+        )
+        lat = self.field.coords[coords[0]].to_numpy()
+        if len(lat.shape) == 1 and lat[-1] < lat[0]:
+            lat = lat[::-1]
+        lon = self.field.coords[coords[-1]].to_numpy()
+        return [lat, lon]
 
     @staticmethod
-    def opposite(values, **kwargs):
-    # pylint: disable=unused-argument
+    def opposite(values: DataArray, **kwargs) -> DataArray:  # noqa: ARG004
+        """Returns the opposite of input values."""
 
-        ''' Returns the opposite of input values  '''
-
-        return - values
+        return -values
 
     @property
-    def valid_dt(self) -> datetime.datetime:
-
-        ''' Returns a datetime object corresponding to the forecast hour's valid
-        time as set in the Grib file. '''
+    def valid_dt(self) -> datetime:
+        """
+        Returns a datetime object corresponding to the forecast hour's valid
+        time as set in the Grib file.
+        """
 
-        fh = datetime.timedelta(hours=int(self.fhr))
+        fh = timedelta(hours=int(self.fhr))
         return self.anl_dt + fh
 
     @abc.abstractmethod
-    def values(self, level=None, name=None, **kwargs):
-
-        ''' Returns the values of a given variable. '''
-        ...
-
-    @staticmethod
-    def vertical_dim(field):
-
-        ''' Determine the vertical dimension of the variable by looking through
-        the field's dimensions for one that includes "lv". Return the first
-        matching instance. '''
-
-        vert_dim = [dim for dim in field.dims if ('lv' in dim or 'probability' in dim)]
-        if vert_dim:
-            return vert_dim[0]
-        return ''
+    def values(
+        self, level: str | None = None, name: str | None = None, do_transform: bool = True
+    ) -> DataArray:
+        """Returns the values of a given variable."""
+
+    def vector_magnitude(
```

Comment:
This was defined nearly the same in both subclasses, so I made some generalizations and moved it to the base class.